src/main/java/org/openshift/jenkins/plugins/openshiftlogin/OpenShiftOAuth2SecurityRealm.java

```diff
@@ -597,6 +605,33 @@ public void initialize(HttpRequest request) throws IOException {
         return transport;
     }
 
+    /**
+     * Since Java 7 and Java 8, it is not required to set a custom Authenticator to
+     * use an authenticated HTTP proxy. However, for HTTPS proxies, this is still
+     * necessary for Basic authentication in conjonction with:
+     *
+     * -Djdk.http.auth.tunneling.disabledSchemes=""
+     * -Djdk.http.auth.proxying.disabledSchemes=""
+     *
+     * If the custom authenticator is not set using (Authenticator.setDefault()) an
+     * error: java.io.IOException: Unable to tunnel through proxy. Proxy returns
+     * "HTTP/1.1 407 Proxy Authentication Required" is thrown.
+     */
+    private void initializeHttpsProxyAuthenticator() {
+        final String username = System.getProperty(HTTPS_PROXY_USER);
+        final String password = System.getProperty(HTTPS_PROXY_PASSWORD);
+        LOGGER.log(INFO, "Checking if HTTPS proxy initialization is required ... ");
+        if (username != null && password != null) {
+            LOGGER.log(INFO, HTTPS_PROXY_USER + " or " + HTTPS_PROXY_PASSWORD + " found in system properties...");
+            LOGGER.log(INFO, "Now setting a custom PasswordAuthentication authenticator");
```

Comment:
nit - first two log statements can probably be set to `DEBUG` - info message will need to be more informative:

```suggestion
            LOGGER.log(INFO, "Creating basic authenticator for HTTPS proxy auth");
```

Reply:
Fixed in forced push commit